tests/v2/validation/snapshot/snapshot.go

```diff
@@ -4,270 +4,351 @@ import (
 
-	deploymentResponse, err := createDeployment(steveclient, initialWorkloadName)
-	require.NoError(t, err)
+	containerTemplate := workloads.NewContainer(containerName, containerImage, corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{}, nil, nil, nil)
+	podTemplate := workloads.NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+	deployment := workloads.NewDeploymentTemplate(initialWorkloadName, defaultNamespace, podTemplate, isCattleLabeled, nil)
 
-	err = workloads.VerifyDeployment(steveclient, deploymentResponse)
-	require.NoError(t, err)
-	require.Equal(t, initialWorkloadName, deploymentResponse.ObjectMeta.Name)
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAppendName + initialWorkloadName,
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name: port,
+					Port: 80,
+				},
+			},
+			Selector: deployment.Spec.Template.Labels,
+		},
+	}
 
-	ingressResp, err := createIngress(steveclient, initialIngressName, serviceAppendName+initialWorkloadName)
+	deploymentResp, serviceResp, err := workloads.CreateDeploymentWithService(steveclient, initialWorkloadName, deployment, service)
 	require.NoError(t, err)
-	require.Equal(t, initialIngressName, ingressResp.ObjectMeta.Name)
 
-	existingSnapshots, err := provisioning.GetSnapshots(client, localClusterID, clusterName)
+	err = workloads.VerifyDeployment(steveclient, deploymentResp)
 	require.NoError(t, err)
+	require.Equal(t, initialWorkloadName, deploymentResp.ObjectMeta.Name)
 
-	etcdNodeCount, _ := MatchNodeToAnyEtcdRole(t, client, clusterID)
+	path := ingresses.NewIngressPathTemplate(networking.PathTypeExact, ingressPath, serviceAppendName+initialWorkloadName, 80)
+	ingressTemplate := ingresses.NewIngressTemplate(initialIngressName, defaultNamespace, "", []networking.HTTPIngressPath{path})
 
-	err = etcdsnapshot.CreateSnapshot(client, clusterName)
+	ingressResp, err := ingresses.CreateIngress(steveclient, initialIngressName, ingressTemplate)
 	require.NoError(t, err)
+	require.Equal(t, initialIngressName, ingressResp.ObjectMeta.Name)
 
-	adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
-	require.NoError(t, err)
+	var existingSnapshots []string
+	var isRKE1 bool
 
-	clusterObject, _, err := clusters.GetProvisioningClusterByName(adminClient, clusterName, namespace)
-	require.NoError(t, err)
+	clusterObject, _, err := clusters.GetProvisioningClusterByName(client, clusterName, namespace)
+	if clusterObject == nil {
+		_, err := client.Management.Cluster.ByID(clusterID)
+		require.NoError(t, err)
 
-	initialKubernetesVersion := clusterObject.Spec.KubernetesVersion
-	steveID := fmt.Sprintf("%s/%s", clusterObject.Namespace, clusterObject.Name)
-	err = clusters.WatchAndWaitForCluster(client, steveID)
-	require.NoError(t, err)
+		isRKE1 = true
```

Comment:
I feel like it would be cleaner to have this call one of 2 helpers:
snapshotRestoreRKE1 vs snapshotRestoreR2K3s, or just depend on the user calling this function. As is, the rest of this function has a little bit too many if statements just to handle the rke1 use cases for readability.

Reply:
Yeah, @igomez06 had talks of this. There isn't really a great way around this problem as we want the snapshot test suite to be able to run on any clusters. There's quite a bit of redundancy if we try to make a separate file or function calls, hence why this is the way it is.

Reply:
Talked offline about this. Restructed this so that this was implemented, I will provide jenkins jobs tomorrow morning.